File: src/main/java/lol/j0/modulus/registry/ToolRegistry.java
package lol.j0.modulus.registry;

import lol.j0.modulus.api.RegisteredTool;
import net.minecraft.item.Item;
import net.minecraft.util.Identifier;

import java.util.HashMap;

public class ToolRegistry {

	static {
		TOOLS = new HashMap<>();
	}
	public static HashMap<Identifier, RegisteredTool> TOOLS;

	public static void register(Identifier identifier, Item item) {
		ToolRegistry.TOOLS.put(identifier, new RegisteredTool(identifier, item));
	}
}

File: src/main/kotlin/lol/j0/modulus/registry/ToolRegistry.kt
package lol.j0.modulus.registry

import lol.j0.modulus.api.RegisteredTool
import net.minecraft.item.Item
import net.minecraft.util.Identifier

object ToolRegistry {
    var TOOLS: HashMap<Identifier, RegisteredTool>? = null

    init {
        TOOLS = HashMap()
    }

    fun register(identifier: Identifier, item: Item?) {
        TOOLS!![identifier] = RegisteredTool(identifier, item!!)
    }
}
